### ISSUE TITLE
sparsetensor-input in the form of ids and values as FeatureColumns for tf.estimator use

### ISSUE THREAD

**Issue by hamidmaei** [tensorflow/tensorflow#26456]
I have been struggling with this issue for a while and in TF Dev Summit 2019, also brought it with @mrry who helped me a lot but still it is a fully unresolved issue.

I have input data in the form of ids and values; that is sparse representation for a giant input feature whose non-zero element indices are stored in ids and corresponding values are stored in values. I would like to use estimators but estimators require feature_column format as high-level APIs to dead with my input and currently I don't see how I can use feature-columns with my data format---please note I already have done all the feature engineering and representations and stored them in my ids, values so no need for bucketizzation, and other feature transformations that feature_columns are used for. Also the inputs are in the form of sparsetensors since I am dealing with a giant feature-vector and sparse representation will save a lot of computation and memory. How can I do it with estimators and go around the ecisting feature_column format?

**Comment by hamidmaei** [comment]
Just in case here is a simple demonstration to show that I can use this sparse input for training without using estimators but when it comes to estimators I don't now how to do it using feature columns.

Say we already have stored tfrecords dataset in a file. The data is stored in the form of dictionary with keysa such as label, ids and values. We generate min-batches after calling .map and .batch according to:

```
iterator= dataset.make_one_shot_iterator()
features=iterator.get_next()
labels =features["label"]
sparse_ids = features["ids"]
sparse_values = features["values"]
```

Now I can call the following function to do dot product needed for a simple linear regression:

```
def sparse_dot_product(sparse_ids,
                          sparse_values,
                          weights_shape,
                          biases_shape,
                          is_train=True):
    weights = tf.get_variable(
        "weights", weights_shape, initializer=tf.initializers.zeros())
    biases = tf.get_variable(
        "biases", biases_shape, initializer=tf.initializers.zeros())
    return tf.nn.embedding_lookup_sparse(
        weights, sparse_ids, sparse_values, combiner="sum") + biases
```

and then can do logistic regression training using tf.nn.sparse_softmax_cross_entropy_with_logits() and tf.reduce_mean(..) functions. Very simple and fast since I am doing lookup! 

Now using estimators is supposed to be easier, but to me it doesn't, due to the current forms of feature_columns. Any help will be appreciated

**Comment by hamidmaei** [comment]
I actually found the solution! custom estimators are the way to implement this using model_fn as follows:

`def my_model_fn(features,labels, mode):
  batch_ids = features['ids']
  batch_values = features['values']
  batch_labels = labels
  logits= your_inference_function(batch_ids, batch_values)

  cross_entropy = tf.nn.sparse_softmax_cross_entropy_with_logits(
                                                              logits=logits, labels=batch_labels
                                                              )
  loss = tf.reduce_mean(cross_entropy, name="loss")
  optimizer = get_your_optimizer , e.g. rmsprop
  train_op = optimizer.minimize(loss, global_step=tf.train.get_global_step())
  tf.get_variable_scope().reuse_variables()
  # Define accuracy op for train data
  train_accuracy_logits = inference(batch_ids, batch_values, True)
  train_softmax = tf.nn.softmax(train_accuracy_logits)
  train_correct_prediction = tf.equal(
                              tf.argmax(train_softmax, 1), batch_labels
                              )
  train_accuracy = tf.reduce_mean(
                              tf.cast(train_correct_prediction, tf.float32)
                              )
  

  return tf.estimator.EstimatorSpec(mode, loss=loss, train_op=train_op)`

then use: 
classifier = tf.estimator.Estimator(
        model_fn=my_model_fn)


classifier.train(
         input_fn=lambda:input_fn(),
         steps=steps_to_run)

**Comment by jvishnuvardhan** [comment]
Thanks for posting solution also. I am closing the issue. Thanks!

**Comment by tddevlin** [comment]
Could we reopen this issue? The solution proposed by @hamidmaei is a legitimate workaround, but it doesn't solve the underlying incompatibility between sparse tensors and feature columns. Using sparse tensor representations of feature counts is a common use case (e.g. bag-of-words models) and it should be possible to plug these sparse tensors directly into a premade Estimator, but it seems like this currently isn't possible with feature columns.

**Comment by jvishnuvardhan** [comment]
@tddevlin Can you please open a new issue and mention about this old issue. Can you please provide a simple standalone code to reproduce the issue? Thanks!
